Framework/PythonInterface/mantid/api/src/Exports/IPeak.cpp

```diff
@@ -165,6 +165,9 @@ void export_IPeak() {
       .def("setGoniometerMatrix", &setGoniometerMatrix, (arg("self"), arg("goniometerMatrix")),
            "Set the :class:`~mantid.geometry.Goniometer` rotation matrix of "
            "this peak.")
+      .def("getDetectorID", &IPeak::getDetectorID, arg("self"),
```

Comment:
Thanks, I've been meaning to expose this to python for some time!
The doc string states that this works only for rectangular detectors, however peak tables in WISH do have the correct DetID populated, so perhaps it works in general?

Reply:
It is not tied to rectangular detectors, I agree.